### PR TITLE
feat: add file parameter to the action

### PR DESCRIPTION
**Pull request by mohammadne**
`path` parameter to the pipeline has been added

### CODE REVIEW

File: action.yml [hunk 1]
```diff
@@ -2,19 +2,22 @@ name: "Okteto Pipeline"
 description: "Trigger your repository's okteto pipeline"
 inputs:
   name:
-    description: 'The name of the pipeline'
```
Comment (jmacelroy) [inline action.yml:5]: What is the reasoning behind changing double quotes to single quotes. I'd prefer to keep them as they are if possible.

File: entrypoint.sh [hunk 2]
```diff
@@ -5,7 +5,8 @@ name=$1
 namespace=$2
 timeout=$3
 skip_if_exists=$4
-variables=$5
+path=$5
+variables=$6
```
Comment (jmacelroy) [inline entrypoint.sh]: I'm not sure of the impact of changing the parameters for the bash script. @ifbyol or @rberrelleza do either of y'all know the impact of doing this?

File: entrypoint.sh [hunk 3]
```diff
@@ -50,5 +51,5 @@ fi
 
 export OKTETO_DISABLE_SPINNER=1
 
-echo running: okteto pipeline deploy --name "${name}" --branch="${branch}" --repository="${GITHUB_SERVER_URL}/${repository}" ${params} --wait
-okteto pipeline deploy --name "${name}" --branch="${branch}" --repository="${GITHUB_SERVER_URL}/${repository}" ${params} --wait
+echo running: okteto pipeline deploy -f "${path}" --name "${name}" --branch="${branch}" --repository="${GITHUB_SERVER_URL}/${repository}" ${params} --wait
```
Comment (jmacelroy) [inline entrypoint.sh]: What happens if there is no path specified? I think we should only be adding the `-f` option if a path is indeed specified.

File: action.yml (hunk 1 above)
Comment (mohammadne) [reply action.yml:5]: as you can see [here](https://github.com/mohammadne/pipeline/blob/master/action.yml) I have used double quotes.

**Comment by mohammadne** [comment]
thanks @jmacelroy 
I have resolve them

File: README.md [hunk 4]
```diff
@@ -35,6 +35,10 @@ Skip the pipeline deployment if the pipeline already exists in the namespace (de
 
 A list of variables to be used by the pipeline. If several variables are present, they should be separated by commas e.g. VAR1=VAL1,VAR2=VAL2,VAR3=VAL3.
 
+### `filePath`
```
Comment (pchico83) [inline README.md]: In the Okteto CLI is named `filename`:
```
  -f, --filename string     relative path within the repository to the manifest file (default to okteto-pipeline.yaml or .okteto/okteto-pipeline.yaml)
```

File: README.md [hunk 5]
```diff
@@ -112,5 +116,6 @@ jobs:
         name: pr-${{ github.event.number }}
         timeout: 8m
         skipIfExists: true
+        path: .okteto/custom-pipeline.yaml
```
Comment (pchico83) [inline README.md]:
```suggestion
        filename: .okteto/custom-pipeline.yaml
```

File: entrypoint.sh [hunk 6]
```diff
@@ -50,5 +51,20 @@ fi
...
+output=$(cat <<-END
+    okteto pipeline deploy
+    --name "${name}"
+    --branch="${branch}"
+    --repository="${GITHUB_SERVER_URL}/${repository}"
+    ${params}
+END
+)
+
+if [ -n "$filePath" ]; then
```
Comment (pchico83) [inline entrypoint.sh]: Could you follow the same approach than other parámeters? that means doing something like:
```
if [ ! -z "$filePath" ]; then
params="${params} -f "${filePath}""
fi
```
also move it before the line `output=...`

File: entrypoint.sh [hunk 7]
```diff
@@ -50,5 +51,20 @@ fi
 
 export OKTETO_DISABLE_SPINNER=1
 
-echo running: okteto pipeline deploy --name "${name}" --branch="${branch}" --repository="${GITHUB_SERVER_URL}/${repository}" ${params} --wait
-okteto pipeline deploy --name "${name}" --branch="${branch}" --repository="${GITHUB_SERVER_URL}/${repository}" ${params} --wait
+output=$(cat <<-END
```
Comment (pchico83) [inline entrypoint.sh]: Could you leave it directly running the command? you can use `\` to break lines for clarity

File: entrypoint.sh [hunk 8]
```diff
@@ -50,5 +51,20 @@ fi
...
+    --repository="${GITHUB_SERVER_URL}/${repository}"
+    ${params}
+END
+)
+
+if [ -n "$filePath" ]; then
+  output="$output -f "${filePath}""
+fi
+
+output="$output --wait"
```
Comment (pchico83) [inline entrypoint.sh]: same, we can move it right to the `okteto pipeline` command

**Comment by mohammadne** [comment]
thanks @pchico83 
resolved !

File: README.md (hunk 5 above)
Comment (pchico83) [inline README.md]:
```suggestion
        filename: .okteto/custom-pipeline.yaml
```

**Review by pchico83 — APPROVED** [review]

File: README.md [hunk 9]
```diff
@@ -35,6 +35,10 @@ Skip the pipeline deployment if the pipeline already exists in the namespace (de
 
 A list of variables to be used by the pipeline. If several variables are present, they should be separated by commas e.g. VAR1=VAL1,VAR2=VAL2,VAR3=VAL3.
 
+### `filename`
+
+path to the pipeline file, if not specified the [default file path](https://okteto.com/docs/cloud/okteto-pipeline#customize-the-okteto-pipeline) will be used
```
Comment (pchico83) [inline README.md]:
```suggestion
The relative path within the repository to the pipeline file. If not specified the [default pipeline](https://okteto.com/docs/cloud/okteto-pipeline#customize-the-okteto-pipeline) is executed
```

File: README.md [hunk 10]
```diff
@@ -35,6 +35,10 @@ Skip the pipeline deployment if the pipeline already exists in the namespace (de
 
 A list of variables to be used by the pipeline. If several variables are present, they should be separated by commas e.g. VAR1=VAL1,VAR2=VAL2,VAR3=VAL3.
 
+### `filename`
+
+The relative path within the repository to the pipeline file. If not specified the [default pipeline](https://okteto.com/docs/cloud/okteto-pipeline#customize-the-okteto-pipeline) is executed
```
Comment (pchico83) [inline README.md]:
```suggestion
The relative path within the repository to the pipeline file. If not specified the [default pipeline](https://okteto.com/docs/cloud/okteto-pipeline#customize-the-okteto-pipeline) is executed.
```

**Review by pchico83 — APPROVED** [review]

**Comment by pchico83** [comment]
@mohammadne thanks for contributing to okteto!

**Pull request merged**